crates/testing/src/sdk.rs

```diff
@@ -34,41 +35,9 @@ pub fn ensure_standalone_process() {
     }
 }
 
-lazy_static! {
-    /// An exclusive lock which ensures we only run `spacetime generate` once for each target directory.
-    ///
-    /// Without this lock, if multiple `Test`s ran concurrently in the same process
-    /// with the same `client_project` and `generate_subdir`,
-    /// the test harness would run `spacetime generate` multiple times concurrently,
-    /// each of which would remove and re-populate the bindings directory,
-    /// potentially sweeping them out from under a compile or run process.
-    ///
-    /// This lock ensures that only one `spacetime generate` process runs at a time,
-    /// and the `HashSet` ensures that we run `spacetime generate` only once for each output directory.
-    ///
-    /// Circumstances where this will still break:
-    /// - If multiple tests want to use the same client_project/generate_subdir pair,
-    ///   but for different modules' bindings, only one module's bindings will ever be generated.
-    ///   If you need bindings for multiple different modules, put them in different subdirs.
-    /// - If multiple distinct test harness processes run concurrently,
-    ///   they will encounter the race condition described above,
-    ///   because the `BINDINGS_GENERATED` lock is not shared between harness processes.
-    ///   Running multiple test harness processes concurrently will break anyways
-    ///   because each will try to run `spacetime start` as a subprocess and will therefore
-    ///   contend over port 3000.
-    ///   Prefer constructing multiple `Test`s and `Test::run`ing them
-    ///   from within the same harness process.
-    //
-    // I (pgoldman 2023-09-11) considered, as an alternative to this lock,
-    // having `Test::run` copy the `client_project` into a fresh temporary directory.
-    // That would be more complicated, as we'd need to re-write dependencies
-    // on the client language's SpacetimeDB SDK to use a local absolute path.
-    // Doing so portably across all our SDK languages seemed infeasible.
-    static ref BINDINGS_GENERATED: Mutex<HashSet<String>> = Mutex::new(HashSet::new());
-}
-
 pub struct Test {
     /// A human-readable name for this test.
+    #[allow(dead_code)] // TODO: should we just remove this now that it's unused?
     name: String,
```

Comment:
Do we not print it when a test fails?

Reply:
No, Cargo uses test names from corresponding functions. Before this PR this field was used in compilation messages, but now that they're memoized and shared between tests, it seemed misleading to use arbitrary test name (whichever happens first) in that context, so I changed them to just messages like "(compiling)".

Now this field is completely unused, but it's assigned in a bunch of spaces and I wasn't sure whether it's worth removing it altogether from all those places or whether we want to use it again in the future.